### PR TITLE
Don't drop to newline when serializing patterns starting with special chars

### PR DESCRIPTION
**Pull request by stasm**
A port of https://github.com/projectfluent/fluent.js/pull/513.

### CODE REVIEW

File: fluent.syntax/src/main/kotlin/org/projectfluent/syntax/serializer/Serializer.kt [hunk 1]
```diff
@@ -2,11 +2,33 @@ package org.projectfluent.syntax.serializer
...
+private fun PatternElement.includesLine() =
+    this is TextElement && value.contains("\n")
+
+private fun PatternElement.isSelectExpr() =
+    this is Placeable && expression is SelectExpression
+
+private fun Pattern.shouldStartOnNewLine(): Boolean {
+    val isMultiline = this.elements.any { it.isSelectExpr() || it.includesLine() }
+    if (isMultiline) {
+        val firstElement = this.elements[0]
```
Comment (Pike) [inline fluent.syntax/src/main/kotlin/org/projectfluent/syntax/serializer/Serializer.kt:17]: Should this use https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/element-at-or-null.html to be safe?

File: fluent.syntax/src/main/kotlin/org/projectfluent/syntax/serializer/Serializer.kt [hunk 2]
```diff
@@ -2,11 +2,33 @@ package org.projectfluent.syntax.serializer
...
+        if (firstElement is TextElement) {
+            val firstChar = firstElement.value[0]
```
Comment (Pike) [inline fluent.syntax/src/main/kotlin/org/projectfluent/syntax/serializer/Serializer.kt:19]: Same here?

**Review by Pike — APPROVED** [review]
LGTM, just two nits that would make this code safe in cases where you programmatically create weird patterns. As I mentioned on the js one, the first is safe. Not sure about the empty text one.

I'd rely on whatever IDEA says in tooltips ;-)

Nice that the check is in one loop, instead of looping twice.  That's a nit I dropped nagging about on the js impl ;-)

File: fluent.syntax/src/main/kotlin/org/projectfluent/syntax/serializer/Serializer.kt (hunk 1 above)
Comment (stasm) [reply fluent.syntax/src/main/kotlin/org/projectfluent/syntax/serializer/Serializer.kt:17]: Ah, nice, I was wondering if I should have somehow guarded this against out-of-bounds. Thanks.

**Pull request merged**